src/Core/Utility/Form.php

```diff
@@ -0,0 +1,454 @@
+            'placeholder' => __('confirm password'),
+            'autocomplete' => 'new-password',
+            'default' => '',
+            'type' => 'password',
+        ];
+    }
+
+    /**
+     * Options for title
+     *
+     * @return array
+     */
+    protected static function titleOptions() : array
+    {
+        return [
+            'class' => 'title',
+            'type' => 'text',
+        ];
+    }
+
+    /**
+     * Get control type per string, according to schema property data.
+     * Possibilities:
+     *
+     *    format 'date-time' => 'date-time'
+     *    format 'date' => 'date'
+     *    contentMediaType => 'textarea'
+     *    schema.enum is an array => 'enum'
+     *
+     * Return 'text' otherwise.
+     *
+     * @param array $schema The property schema
+     * @return string
+     */
+    protected static function typeFromString(array $schema) : string
+    {
+        if (!empty($schema['format']) && in_array($schema['format'], ['date', 'date-time'])) {
+            return $schema['format'];
+        }
+        if (!empty($schema['contentMediaType']) && $schema['contentMediaType'] === 'text/html') {
+            return 'textarea';
+        }
+        if (!empty($schema['enum']) && is_array($schema['enum'])) {
+            return 'enum';
+        }
+
+        return 'text';
+    }
+
+    /**
+     * Return the type for number: 'number'
+     *
+     * @param array $schema Object schema array.
+     * @return string
+     */
+    protected static function typeFromNumber(array $schema) : string
+    {
+        return 'number';
+    }
+
+    /**
+     * Return the type for integer: 'number'
+     *
+     * @param array $schema Object schema array.
+     * @return string
+     */
+    protected static function typeFromInteger(array $schema) : string
+    {
+        return 'number';
+    }
+
+    /**
+     * Return the type for boolean: 'checkbox'
+     *
+     * @param array $schema Object schema array.
+     * @return string
+     */
+    protected static function typeFromBoolean(array $schema) : string
+    {
+        return 'checkbox';
+    }
+
+    /**
+     * Return the type for array: 'checkbox'
+     *
+     * @param array $schema Object schema array.
+     * @return string
+     */
+    protected static function typeFromArray(array $schema) : string
+    {
+        return 'checkbox';
+    }
+
+    /**
+     * Return the type for object: 'json'
+     *
+     * @param array $schema Object schema array.
+     * @return string
+     */
+    protected static function typeFromObject(array $schema) : string
+    {
+        return 'json';
+    }
+
+    /**
+     * Return method [Form::class, $methodName], if it's callable.
+     * Otherwise, throw \InvalidArgumentException
+     *
+     * @param string $name The method name
+     * @return array
+     */
+    public static function getMethod(string $name) : array
+    {
+        $methodName = Inflector::variable(str_replace('-', '_', $name));
```

Comment:
[reminder] we should rename `confirm-password` as `confirm_password` - convention is snake case, no hyphens